lib/mongoid/scroll/base64_encoded_cursor.rb

```diff
@@ -0,0 +1,29 @@
+require 'base64'
+require 'json'
+
+module Mongoid
+  module Scroll
+    # Allows to serializer/deserialize the cursor using RFC 4648
+    class Base64EncodedCursor < Cursor
+      def to_s
+        Base64.strict_encode64({ value: super, field_type: field_type, field_name: field_name, direction: direction }.to_json)
+      end
+
+      class << self
+        def deserialize(str)
+          config_hash = ::JSON.parse(::Base64.strict_decode64(str))
+          new(config_hash['value'], field_type: config_hash['field_type'], field_name: config_hash['field_name'], direction: config_hash['direction'])
+        rescue ArgumentError
+          raise Mongoid::Scroll::Errors::InvalidBase64Error.new(str: str)
```

Comment:
My goal was mainly to not raise the error directly from `Base64.strict_decode64` if the Base64 string was invalid:
```ruby
[1] pry(main)> Base64.strict_decode64('foo')
ArgumentError: invalid base64
from <internal:pack>:281:in `unpack1'
```

But here it also wraps `new` which can also raise an `ArgumentError` from `Mongoid::Scroll::Cursor.extract_field_options`.

I don't really know what's best:

1. Wrapping only `Base64.strict_decode64` to have a custom exception instead of the original one
2. Wrapping everything, in which case an `InvalidBase64CursorError` probably makes more sense
3. Not rescue at all, and let the original exception (from `Base64`, `JSON`, `Mongoid::Scroll::Cursor` or anything else) bubble up

I think I'd go with 3. and just remove the rescue/not create a custom exception.

Reply:
Understood. I think you should raise an `InvalidCursorError` just like [here](https://github.com/mongoid/mongoid-scroll/blob/master/lib/mongoid/scroll/cursor.rb#L71). If there's additional information and you want to include something specific about the base64 error, subclass `InvalidCursorError` into `InvalidBase64CursorError` (this way anyone can rescue any `InvalidCursorError` like before). I was mostly commenting on the naming.

Reply:
Done in 2415bb3bc944f6d0aa354bed5955c82028cd90d1.